cpp/ql/test/experimental/query-tests/Security/CWE/CWE-415/semmle/tests/test.c

```diff
@@ -0,0 +1,21 @@
+typedef unsigned long size_t;
+void *malloc(size_t size);
+void free(void *ptr);
+
+void workFunction_0(char *s) {
+  int intSize = 10;
+  char *buf;
+  buf = (char *) malloc(intSize);
+  free(buf); // BAD
+  if(buf) free(buf);
```

Comment:
This `if(buf)` is irrelevant. IMHO you should make two tests: with `if` and without to demonstrate it doesn't influence the analysis.